app/lib/submission_builder/state_return.rb

```diff
@@ -0,0 +1,96 @@
+module SubmissionBuilder
+  class StateReturn < SubmissionBuilder::Document
+    def document
+      @document = state_schema_version.present? ?
+                    build_xml_doc(build_xml_doc_tag, stateSchemaVersion: state_schema_version) :
+                    build_xml_doc(build_xml_doc_tag)
+      build_headers
+      build_main_document
+      build_documents
+      build_state_specific_tags(@document)
+      @document
+    end
+
+    def pdf_documents
+      included_documents.map { |item| item if item.pdf }.compact
+    end
+
+    def build_headers
+      @document.at("ReturnState").add_child(authentication_header)
+      @document.at("ReturnState").add_child(return_header)
+    end
+
+    def build_main_document
+      @document.at("ReturnState").add_child("<ReturnDataState></ReturnDataState>")
+    end
+
+    def build_documents
+      @document.at("ReturnDataState").add_child(documents_wrapper)
+      attached_documents.each do |attached|
+        @document.at(attached_documents_parent_tag).add_child(document_fragment(attached))
+      end
+    end
+
+    def document_fragment(document)
+      document[:xml_class].build(@submission, validate: false, kwargs: document[:kwargs]).document.at("*")
+    end
+
+    def authentication_header
+      SubmissionBuilder::AuthenticationHeader.build(@submission, validate: false).document.at("*")
+    end
+
+    def return_header
+      SubmissionBuilder::ReturnHeader.build(@submission, validate: false).document.at("*")
+    end
+
+    def attached_documents
+      @attached_documents ||= xml_documents.map { |doc| { xml_class: doc.xml, kwargs: doc.kwargs } }
+    end
+
+    def xml_documents
+      included_documents.map { |item| item if item.xml }.compact
+    end
+
+    def included_documents
+      supported_documents.map { |item| OpenStruct.new(**item, kwargs: item[:kwargs] || {}) if item[:include] }.compact
+    end
+
+    def combined_w2s
+      @submission.data_source.direct_file_data.w2s.map.with_index do |w2, i|
+        intake = @submission.data_source
+        intake_w2 = intake.state_file_w2s.find { |w2| w2.w2_index == i } if intake.state_file_w2s.present?
+
+        {
+          xml: SubmissionBuilder::Shared::ReturnW2,
```

Comment:
Looks like `ReturnW2` is the only class in `Shared` that isn't CTC related. Do you think it would make sense to put `StateReturn` et al into shared, or pull `ReturnW2` out? Either way I'd like them to be in the same directory.

Reply:
I moved everything in submission_builder/shared/ up one level to submission_builder/